### PR TITLE
tempo: 1.1.0 -> 1.4.1, add NixOS module

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Catching up on nearly a year of updates. Supersedes #178249 because I broke Github.
Changelogs: https://github.com/grafana/tempo/releases

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/tracing/tempo.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  settingsFormat = pkgs.formats.yaml {};
+in {
+  options.services.tempo = {
+    enable = mkEnableOption "Grafana Tempo";
+
+    settings = mkOption {
+      type = settingsFormat.type;
+      default = {};
+      description = ''
+        Specify the configuration for Tempo in Nix.
```
Comment (SuperSandro2000) [inline nixos/modules/services/tracing/tempo.nix:17]: Is there an upstream doc we can link for options?

Comment (K900) [reply nixos/modules/services/tracing/tempo.nix:17]: Yeah, added.

File: nixos/modules/services/tracing/tempo.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      in
+      {
+        ExecStart = "${pkgs.tempo}/bin/tempo --config.file=${conf}";
+        DynamicUser = true;
+        Restart = "always";
+        ProtectSystem = "full";
+        DevicePolicy = "closed";
+        NoNewPrivileges = true;
+        WorkingDirectory = "/var/lib/tempo";
+        StateDirectory = "tempo";
```
Comment (SuperSandro2000) [inline nixos/modules/services/tracing/tempo.nix:64]: Some more hardening wouldn't hurt.

**Review by Mindavi — APPROVED** [review]
Diff LGTM, not very well versed in modules though

File: nixos/modules/services/tracing/tempo.nix (hunk 2 above)
Comment (K900) [reply nixos/modules/services/tracing/tempo.nix:64]: This is already better than upstream: https://github.com/grafana/tempo/blob/main/tools/packaging/tempo.service

**Comment by K900** [comment]
I'm just going to merge it and experiment with some more hardening options for the next update.

**Pull request merged**